packages/enzyme/src/ShallowWrapper.js

```diff
@@ -1033,6 +1033,34 @@ class ShallowWrapper {
     return this.props()[propName];
   }
 
+  /**
+   * Returns a wrapper of the node rendered by the provided render prop.
+   *
+   * @param {String} propName
+   * @returns {ShallowWrapper}
+   */
+  renderProp(propName, ...args) {
+    const Wrapper = props => props.children;
+    return this.single('renderProp', () => {
+      if (typeof propName !== 'string') {
+        throw new TypeError('`propName` must be a string');
+      }
+      const prop = this.prop(propName);
+      if (!prop) {
+        throw new Error(`no prop called “${propName}“ found`);
+      }
+      if (typeof prop !== 'function') {
+        throw new Error(
+          `expected prop “${propName}“ to contain a function, but it holds “${typeof prop}“`,
+        );
+      }
+
+      const element = prop(...args);
+      const wrapped = getAdapter(this[OPTIONS]).createElement(Wrapper, null, element);
```

Comment:
we may need to add a method to all the adapters "wrap", so that the adapter itself can handle the wrapper implementation.

Reply:
I think it'd need to be in this PR (in a separate commit, ofc).